game/src/js/util/Controller.js

```diff
@@ -1,19 +1,53 @@
 export default class Controller {
   constructor(scene) {
     this.cursors = scene.input.keyboard.createCursorKeys();
-
+    this.keyW = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.W);
+    this.keyA = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.A);
+    this.keyS = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.S);
+    this.keyD = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.D);
     
   }
 
   getWASDCoordinate() {
     let direction = this.cursors.down.isDown ? 'S' : (this.cursors.up.isDown ? 'N' : '');
     direction += this.cursors.left.isDown ? 'W' : (this.cursors.right.isDown ? 'E' : '');
+
+    direction += this.keyA.isDown ? 'W' : (this.keyD.isDown ? 'E' : '');
+    direction += this.keyW.isDown ? 'N' : (this.keyS.isDown ? 'S' : '');
     return direction;
   }
 
   getWASDVector() {
-    let x = this.cursors.left.isDown ? -1 : (this.cursors.right.isDown ? 1 : 0);
-    let y = this.cursors.up.isDown ? -1 : (this.cursors.down.isDown ? 1 : 0);
+    let x = 0;
+    let y = 0;
+
+    if (this.cursors.up.isDown) {
+      y = -1;
+
+    }
+    if (this.cursors.down.isDown) {
+      y += 1;
+    }
+    if (this.keyW.isDown) {
+      y += -1;
```

Comment:
If both the up arrow and W are pressed, the y value would be -2. This is totally an edge case, and will be fixed by addressing the comment above.